src/error.js

```diff
@@ -536,7 +536,8 @@ export function getErrorReportData(message, filename, line, col, error,
  * @visibleForTesting
  */
 export function detectNonAmpJs(win) {
-  const scripts = win.document.querySelectorAll('script[src]');
+  const scripts = Services.ampdocServiceFor(win)
+      .getAmpDoc().getHeadNode().querySelectorAll('script[src]');
```

Comment:
Did you test that this works as expected? I think you need to pass a node inside the shadow DOM for the PWA case, but I'm not sure how to get one here...